xbmc/SystemGlobals.cpp

```diff
@@ -35,6 +35,10 @@
 #include "interfaces/python/XBPython.h"
 #endif
 
+// Guarantee that CSpecialProtocol is initialized before and uninitialized after RarManager
+#include "filesystem/SpecialProtocol.h"
+std::map<std::string, std::string> CSpecialProtocol::m_pathMap;
```

Comment:
Yep, moving it to .h doesn't compile (and it's then also less readable I think). Long term solution would be to get rid of such globals.